src/node_crypto.cc

```diff
@@ -3767,9 +3767,18 @@ void CipherBase::SetAuthTag(const FunctionCallbackInfo<Value>& args) {
     return args.GetReturnValue().Set(false);
   }
 
-  // FIXME(bnoordhuis) Throw when buffer length is not a valid tag size.
+  unsigned int tag_len = Buffer::Length(args[0]);
+  if (EVP_CIPHER_CTX_mode(cipher->ctx_) == EVP_CIPH_GCM_MODE) {
```

Comment:
Superfluous, always true because of the `cipher->IsAuthenticatedMode()` check above.

Reply:
If we ever add support for other authenticated modes, `IsAuthenticatedMode` will be true even if the cipher mode is not GCM, and other modes might have different restrictions on the tag size. I can remove the check if you would like.

Reply:
We'd audit all `IsAuthenticatedMode()` call sites anyway in that case so that's not a problem.